hwtracer/src/collect/mod.rs

```diff
@@ -22,43 +22,16 @@ static PERF_DFLT_AUX_BUFSIZE: LazyLock<size_t> = LazyLock::new(|| {
 
 const PERF_DFLT_INITIAL_TRACE_BUFSIZE: size_t = 1024 * 1024; // 1MiB
 
-thread_local! {
-    /// When `Some` holds the `ThreadTraceCollector` that is collecting a trace of the current
-    /// thread.
-    static THREAD_TRACE_COLLECTOR: RefCell<Option<Box<dyn ThreadTracer>>> = RefCell::new(None);
```

Comment:
Aren't we going to end up bringing this back later to get back the "only one tracer per thread" property?

Reply:
I don't see it.

If you search for `thread_local!` in the diff for this whole PR, you can see that it only appears in deleted lines.

@ptersilie and I have been debugging the branch and the problem appears to be a thread starts tracing when it is already tracing. This manifests as a "trace" which starts fine, but is later interrupted by the start of another trace. The trace compiler is then confused by the impossible control flow.

You need the thread local (or equivalent) to track whether the current thread is already tracing.